File: demo-iys-rest-scraping.java/src/main/java/demo/model/Target.java
package demo.model;

import com.google.api.client.util.Key;

public class Target implements Comparable<Target> {

    @Key("id")
    private final String id;

    @Key("name")
    private final String name;

    public Target(String id, String name) {
        this.id = id;
        this.name = name;
    }

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o)
            return true;
        if (o == null || getClass() != o.getClass())
            return false;

        Target that = (Target) o;
        return name.equals(that.name);
    }

    @Override
    public int hashCode() {
        return name.hashCode();
    }

    @Override
    public int compareTo(Target that) {
        return this.getName().compareTo(that.getName());
    }
}

File: demo-iys-rest-scraping.kotlin/src/main/kotlin/demo/model/Target.kt
package demo.model

import com.google.api.client.util.Key

class Target(

    @Key("id")
    val id: String,

    @Key("name")
    val name: String) : Comparable<Target> {

    override fun equals(other: Any?): Boolean {
        if (this === other)
            return true
        if (other?.javaClass != javaClass)
            return false

        other as Target
        if (name != other.name) return false

        return true
    }

    override fun hashCode(): Int {
        return name.hashCode()
    }

    override fun compareTo(other: Target): Int {
        return this.name.compareTo(other.name)
    }
}
